perfmetrics/scripts/load_tests/python/load_test.py

```diff
@@ -0,0 +1,357 @@
+# Copyright 2023 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Script to run load tests with given tasks and configuration.
+
+The script requires tasks to run in load tests which can be written in a python
+module or YAML file and then path of that file can be passed to this load script
+using --tasks-python-file-path or --tasks-yaml-file path flag.
+Apart from the tasks, the load test can also be customized in multiple ways
+like number of processes, threads inside each process and run time.
+To know more about the flags supported in script, run:
+  python3 load_test.py --help
+
+Example:
+  python3 load_test.py --task-python-file-path tasks/python_os.py
+  --num-processes 40 --num-threads 1 --run-time 90 --output-dir ~/output
+"""
+import logging
+import importlib
+import importlib.machinery
+import importlib.util
+import inspect
+import sys
+import os
+import time
+import argparse
+import yaml
+import re
+
+from load_generator import load_generator as lg
+from load_generator import task
+from load_generator import constants as lg_const
+from tasks import python_os
+from tasks import tf_data
+from tasks import tf_gfile
+
+READ_WRITE_TASK_TYPES = [
+    python_os.PYTHON_OS_READ, tf_data.TF_DATA_READ, tf_gfile.TF_GFILE_READ
+]
+KNOWN_TASK_TYPES = READ_WRITE_TASK_TYPES
+SIZE_ABBREV_TO_SIZE = {'K': lg_const.KB, 'M': lg_const.MB, 'G': lg_const.GB}
+
+
+def parse_args():
+  """Parses the command line arguments and returns them.
+
+  Returns:
+    Arguments passed through command line set as attributes in args object.
+  """
+  parser = argparse.ArgumentParser(
+      description='Script to do Load testing with '
+      'given task using multiprocessing and '
+      'multithreading on CPU and '
+      'other resources.',
+      formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+  group = parser.add_mutually_exclusive_group(required=True)
+  group.add_argument(
+      '--tasks-python-file-path',
+      type=str,
+      help='Path to python module (file) containing task classes implementing '
+      'the base task.LoadTestTask')
+  group.add_argument(
+      '--tasks-yaml-file-path',
+      type=str,
+      help='Path to yaml file containing configs for known (predefined) '
+      f'tasks: {KNOWN_TASK_TYPES}')
+  parser.add_argument(
+      '--task-names',
+      type=str,
+      default='',
+      help='Comma separated name(s) of tasks (task.LoadTestTask.task_name) in '
```

Comment:
It would be good if you will add an example with value of --task-names, the correct accepted format. This is just to minimize the formatting issue.